framework/src/materials/ParsedMaterialHelper.C

```diff
@@ -27,11 +27,30 @@ ParsedMaterialHelper::ParsedMaterialHelper(const InputParameters & parameters,
                                            VariableNameMappingMode map_mode)
   : FunctionMaterialBase(parameters),
     FunctionParserUtils(parameters),
-    _variable_names(_nargs),
+    _variable_names(0),
     _mat_prop_descriptors(0),
     _tol(0),
-    _map_mode(map_mode)
+    _map_mode(map_mode),
+    variable_info(0)
 {
+  switch (map_mode)
+  {
+    case USE_MOOSE_NAMES:
+      _equation_args = _arg_names;
+      break;
+    case USE_PARAM_NAMES:
+      _equation_args = _arg_param_names;
+      break;
+  }
+  _args.empty();
+  for (unsigned int i = 0; i < _equation_args.size(); ++i)
+  {
+    if (!_arg_param_unique[i] && _map_mode == USE_PARAM_NAMES)
+    {
+      _equation_args[i] += std::to_string(_arg_param_numbers[i]);
+      _arg_names[i] += std::to_string(_arg_param_numbers[i]);
```

Comment:
We should be using `Moose::stringfiy` from `Conversion.h` here (see #14593 and #14594)